### ISSUE TITLE
NamedTuple should be considered immutable (B008)

### ISSUE THREAD

**Issue by s-banach** [astral-sh/ruff#16536]
### Summary

The following triggers B008. I believe it should not?

```python
from typing import NamedTuple


class MyTuple(NamedTuple):
    x: int = 10


def f(m: MyTuple = MyTuple()): ...
```

```
Do not perform function call `MyTuple` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable
```

### Version

ruff 0.9.9 (091d0af2a 2025-02-28)

**Comment by ntBre** [comment]
As a workaround, you may be able to add `MyTuple` to the [`extend-immutable-calls`](https://docs.astral.sh/ruff/settings/#lint_flake8-bugbear_extend-immutable-calls) option, but I'm having some trouble getting that to work within a single script [^1].

However, if I have a `mytuple.py` file:

```python
from typing import NamedTuple


class MyTuple(NamedTuple):
    x: int = 10
```

and a `try.py` script:

```py
from types import MyTuple

def f(m: MyTuple = MyTuple()): ...
```

Then the override does work:

```shell
$ ruff check try.py --select B008 --config 'lint.flake8-bugbear.extend-immutable-calls = ["mytuple.MyTuple"]'
All checks passed!
```
[^1]: We might want to use something other than `from_dotted_name` [here](https://github.com/astral-sh/ruff/blob/main/crates/ruff_linter/src/rules/flake8_bugbear/rules/function_call_in_argument_default.rs#L138)

**Comment by posita** [comment]
Unless I'm mistaken, the subclass work-around from https://github.com/astral-sh/ruff/issues/16536#issuecomment-2704397116 no longer works:

```py
from contextvars import ContextVar
from typing import NamedTuple

class _Context(NamedTuple):
    a: int = 1
    b: bool = True

_ctxt: ContextVar[_Context] = ContextVar(
    "TEST_CONTEXT",
    default=_Context(),  # <- shouldn't a NamedTuple be considered immutable?
)
```

[Playground](https://play.ruff.rs/079df7ce-a20a-4fbd-8cae-af150f645265)

**Comment by ntBre** [comment]
You have to set `lint.flake8-bugbear.extend-immutable-calls` to the full path to the class: https://play.ruff.rs/d8fccf58-a3f4-4c0b-9cf4-81cd26419b15

This is especially awkward in the playground because the file and module name is internally set to `<filename>` but should look a bit less weird in a real project.
